module/zfs/dmu_zfetch.c

```diff
@@ -369,6 +353,10 @@ dmu_zfetch(zfetch_t *zf, uint64_t blkid, uint64_t nblks, boolean_t fetch_data,
 	 * start just after the block we just accessed.
 	 */
 	pf_start = MAX(zs->zs_pf_blkid, end_of_access_blkid);
+	if (zs->zs_pf_blkid1 < end_of_access_blkid)
+		zs->zs_pf_blkid1 = end_of_access_blkid;
```

Comment:
[nit] I could go either way, but sometimes we use this idiom:
`zs->zs_pf_blkid1 = MAX(zs->zs_pf_blkid1, end_of_access_blkid);`

Reply:
While it is slightly more compact, I think it is arguable what is more readable.